configure.ac

```diff
@@ -220,13 +220,37 @@ AC_ARG_WITH([cpu],
 ],[
   case "$host_cpu" in
   powerpc*)
+    # Test successively newer processors and select the
+    # best implementation.
     AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
       #ifdef _ARCH_PWR6
       # define SUCCESS 1
       #else
       # error "Support for hardware dfp isn't available"
       #endif
-      ]])], [with_dfp=yes], [with_dfp=no]) ;;
+      ]])], [with_dfp=yes; submachine=power6], [with_dfp=no])
+    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+      #ifdef _ARCH_PWR6X
+      # define SUCCESS 1
+      #else
+      # error "Support for hardware dfp isn't available"
+      #endif
+      ]])], [with_dfp=yes; submachine=power6x], [])
+    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+      #ifdef _ARCH_PWR7
```

Comment:
Yes, the order here matters. We test for increasingly newer processors until we fail.

Suggestions for a more canonical way to test for these values? Ideally, we might just run one test which would print out the correct values based on the supplied macros.